cassovary-core/src/main/scala/com/twitter/cassovary/util/collections/CQueue.scala

```diff
@@ -0,0 +1,137 @@
+package com.twitter.cassovary.util.collections
+
+import it.unimi.dsi.fastutil.PriorityQueue
+import it.unimi.dsi.fastutil.ints.{IntArrayFIFOQueue, IntComparator, IntHeapPriorityQueue,
+  IntPriorityQueue}
+import it.unimi.dsi.fastutil.objects.{ObjectArrayFIFOQueue, ObjectArrayPriorityQueue}
+
+/**
+ * Specialized implementation of queue.
+ *
+ * Implemented methods are not boxed, when using with primitive types.
+ */
+abstract class CQueue[@specialized(Int, Long) T] {
+  def first(): T
+  def +=(elem: T)
```

Comment:
oh, it depends on queue implementation I guess. Sorry. Might be worth adding a comment.